src/reactviews/pages/QueryResult/table/tableDataView.ts

```diff
@@ -161,29 +167,83 @@ export class TableDataView<T extends Slick.SlickData>
         }
         this._currentColumnFilters = columns!;
         this._data = this._filterFn!(this._allData, columns!);
+        if (this._resetSortData.length > 0) {
+            this._resetSortData = this._filterFn!(
+                this._resetSortData,
+                columns!,
+            );
+        }
         if (this._data.length === this._allData.length) {
-            this.clearFilter();
+            await this.clearFilter();
         } else {
             console.log("filterstatechange");
             // this._onFilterStateChange.fire();
         }
     }
 
-    public clearFilter() {
+    public async clearFilter() {
         if (this._filterEnabled) {
             this._data = this._allData;
+            if (this._resetSortData.length > 0) {
+                this._resetSortData = new Array(...this._allData);
+            }
             this._allData = [];
             this._filterEnabled = false;
+            if (this._currentColumnSort) {
+                this._data = this._sortFn!(
+                    {
+                        sortCol: this._currentColumnSort.column,
+                        sortAsc:
+                            this._currentColumnSort.sortDirection ===
+                            "sort-asc",
+                        grid: undefined,
+                        multiColumnSort: false,
+                    },
+                    this._data,
+                );
+            }
             // this._onFilterStateChange.fire();
         }
     }
 
     async sort(args: Slick.OnSortEventArgs<T>): Promise<void> {
+        if (this._resetSortData.length === 0) {
+            this._resetSortData.push(...this._data);
+        }
+
         this._data = this._sortFn!(args, this._data);
+        this._currentColumnSort = {
+            column: args.sortCol!,
+            sortDirection: args.sortAsc ? "sort-asc" : "sort-desc",
+        };
         console.log(args);
         // this._onSortComplete.fire(args);
     }
 
+    // TODO: need to consider multiple scenarios:
+    // 1. filter is enabled then sort & unsort,
+    // 2. filter is not enabled, sort, then enable filter & unsort
+    // 3. filter is enabled then sort, and then disabled <--TODO:  doesn't work right nwo
```

Comment:
typo

Reply:
Can you create:
1) one issue for this feature so you can link your PR
2) another issue for tracking the TODO scenerio